### DISCUSSION TITLE
Why not "timestamp with timezone" for timestamp columns in database?

### DISCUSSION THREAD

**Discussion by emrahcom** [ory/kratos#1965] [Q&A] [answered]
Is there any reason not to use `timestamp with timezone` for timestamp columns in database? I'm asking especially for postgres. As you can estimate, `timestamp without timezone` can cause confusion in some cases.

**Comment by aeneasr** [comment, marked as answer]
Timestamps are always UTC when stored in the database, iirc

**Reply by emrahcom** [reply]
AFAIK, `timestamp` stores the localtime without caring timezone in `postgresql`. Therefore there is no way to tell what is the exact time without knowing the system timezone.

But if the column type is `timestamp with timezone`, stored value gives all we need.

**Reply by aeneasr** [reply]
Golang takes care of it though :) Do you have a concrete bug or reproducible issue?

**Reply by emrahcom** [reply]
In my current setup (`v0.8.0-alpha.3`) when I create a new account using browser flow, `identities` table stores the times using localtime (which is `+03` in my `postgresql` server).

This is not a Docker setup. Kratos process and postgresql run on their own standalone server.

**Reply by emrahcom** [reply]
If you have a Dockerized Postgresql server, maybe it uses "`Etc/UTC` by default

**Reply by aeneasr** [reply]
I see, the PG driver we have and all the time tools we use assume UTC when no timezone is set. I'd recommend setting the schema or session timezone to UTC when using Ory Kratos. There's currently no better workaround unfortunately. The DBAL we use currently does not support creating `timestampz`.

**Reply by emrahcom** [reply]
Thank @aeneasr,

I decided to set 'Etc/UTC' as timezone for Kratos server and for  Postgresql server. Also I have another custom table which has a `1-1`  relation with Kratos's `identities` table. I added a `timestamptz` column in it, my app is using this column now
